module2.js

```diff
@@ -98,8 +120,10 @@ function question5() {
 */
 
 function question6(promise) {
-  // TODO
-  return /* ??? */
+  return promise.then(function () {
+    "Ilove" + value
```

Comment:
This isn't correct - the test setup was broken, so you couldn't see it was failing. You need to _return_ the new string for it to be received by the rest of the promise chain, and the content of the string doesn't quite match the expectations.